pkg/cmd/cli/cmd/newapp.go

```diff
@@ -99,6 +101,16 @@ func NewCmdNewApplication(fullName string, f *clientcmd.Factory, out io.Writer)
 				glog.Fatalf("Error: %v", err)
 			}
 
+			label := cmdutil.GetFlagString(c, "label")
+			if len(label) == 0 {
+				label = util.GetLabel()
```

Comment:
As I mentioned below I don't like generating random labels here.  We should attempt to create a label that combines input the user provides in a predictable way.  The console requires users to provide a name that becomes a primary label - we should be doing something compatible with that.  If the user doesn't ask for it, I'm not sure we should add it though.